src/parser/smt2/smt2.h

```diff
@@ -257,6 +261,14 @@ class Smt2 : public Parser
    */
   const LogicInfo& getLogic() const { return d_logic; }
 
+  /**
+   * Get the sygus grammars.
+   */
+  std::vector<std::unique_ptr<api::Grammar>>& getAllocGrammars()
```

Comment:
I would prefer this not to be a public method.

Instead, I would prefer having a method:
`Grammar * mkGrammar();`
which adds to the internal vector and then returns its raw pointer, which is then used throughout Smt2.g.

Reply:
Done! Though I named it `addGrammar`